examples/demo.sh

```
@@ -0,0 +1,463 @@
+
+# Color definitions using ANSI escape codes
+readonly RED='\033[0;31m'
+readonly GREEN='\033[0;32m'
+readonly YELLOW='\033[0;33m'
+readonly BLUE='\033[0;34m'
+readonly PURPLE='\033[0;35m'
+readonly CYAN='\033[0;36m'
+readonly RESET='\033[0m'
+
+# Global variables
+readonly SCRIPT_NAME="$(basename "${BASH_SOURCE[0]}")"
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)